saw-central/src/SAWCentral/Crucible/Common/Pred.hs

```diff
@@ -0,0 +1,27 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module SAWCentral.Crucible.Common.Pred (
+    checkBooleanType
+    ) where
+
+import Cryptol.TypeCheck.Type (tIsBit)
+import CryptolSAWCore.TypedTerm (mkTypedTerm, ttType, ttIsMono)
+import SAWCore.SharedTerm (SharedContext, Term)
+
+-- | Ensure that the term has Cryptol type `Bit`
+checkBooleanType :: SharedContext -> Term -> IO ()
+checkBooleanType sc tm = do
+  tt <- mkTypedTerm sc tm
+  case ttIsMono (ttType tt) of
+    Just ty | tIsBit ty -> pure ()
+            | otherwise -> fail $ "Expected type Bit, got: " ++ show ty
```

Comment:
Calling `show` on a `Term` isn't what you want to do here, as that will lead to verbose, user-unfriendly output like `TCon (TC TCFun) [TCon (TC (TCTuple 0)) [],TCon (TC TCBit) []]`. Instead, I think you want something like:

```suggestion
            | otherwise -> fail $ "Expected type Bit, got: " ++ show (ppTerm PrecNone ty)
```

Where `ppTerm` is a function that pretty-prints a `Term` to output that looks closer to what would appear in the SAWScript source syntax.